File: my-angular-app/src/app/models/Message.ts
import { User } from "./User";

export class Message {
    id: number;
    content: string;
    timestamp: Date;
    sender: User;
    receiver: User;

    constructor(id: number, content: string, timestamp: Date, sender: User, receiver: User) {
      this.id = id;
      this.content = content;
      this.timestamp = timestamp;
      this.sender = sender;
      this.receiver = receiver;
    }
  }

File: exampleapp/src/main/java/com/example/exampleapp/model/Message.java
package com.example.exampleapp.model;

import java.time.LocalDateTime;

import com.fasterxml.jackson.annotation.JsonFormat;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.Getter;
import lombok.Setter;

@Entity
public class Message {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    @Getter
    @Setter
    private String content;

    @Getter
    @Setter
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
    private LocalDateTime timestamp;

    @ManyToOne
    @Getter
    @Setter
    @JoinColumn(name = "sender_id", referencedColumnName = "id", nullable = false)
    private User sender;

    @ManyToOne
    @Getter
    @Setter
    @JoinColumn(name = "receiver_id", referencedColumnName = "id", nullable = false)
    private User receiver;

    // Getters, Setters, Constructors
}
